components/locid/src/parser/langid.rs

```diff
@@ -159,16 +161,20 @@ pub const fn parse_language_identifier_with_single_variant_from_iter(
                 region = Some(s);
                 position = ParserPosition::Variant;
             } else if let Ok(v) = subtags::Variant::from_bytes_manual_slice(t, start, end) {
+                // We cannot handle multiple variants in a const context
+                debug_assert!(variant.is_none());
                 variant = Some(v);
-                break;
+                position = ParserPosition::Variant;
             } else if matches!(mode, ParserMode::Partial) {
                 break;
             } else {
                 return Err(ParserError::InvalidSubtag);
             }
         } else if let Ok(v) = subtags::Variant::from_bytes_manual_slice(t, start, end) {
+            // We cannot handle multiple variants in a const context
+            debug_assert!(matches!(position, ParserPosition::Variant));
+            debug_assert!(variant.is_none());
```

Comment:
Issue: This _should_ return an error if `variant.is_none()` in this spot.

I think tests didn't catch this because they don't know the difference between `panic` and `debug_assert`. If you were to run some sort of test in release (non-debug) mode, you would get incorrect behavior.

Reply:
Makes sense. Fixed.